src/hooks/useDropzoneArea.js

```diff
@@ -0,0 +1,80 @@
+import {useState, useEffect, useCallback} from 'react';
+import {createFileFromUrl, readFile} from '../helpers';
+
+export const useDropzoneArea = ({
+    onChange,
+    clearOnUnmount,
+    initialFiles,
+    filesLimit,
+    onDelete}) => {
+    const [fileObjects, setFileObjects] = useState([]);
+
+    useEffect(() => {
+        if (onChange) {
+            onChange(fileObjects.map((fileObject) => fileObject.file));
+        }
+    }, [fileObjects, onChange]);
+
+    useEffect(() => {
+        loadInitialFiles();
+
+        return () => {
+            if (clearOnUnmount) {
+                setFileObjects([]);
+            }
+        };
+    }, [loadInitialFiles]);
+
+    const loadInitialFiles = useCallback(async() => {
+        try {
+            const fileObjs = await Promise.all(
+                initialFiles.map(async(url) => {
+                    const file = await createFileFromUrl(url);
+                    const data = await readFile(file);
+                    return {
+                        file,
+                        data,
+                    };
+                })
+            );
+
+            setFileObjects((prev) => ([...prev, ...fileObjs]));
+        } catch (err) {
+            console.log(err);
```

Comment:
Should probably throw an actual error here rather than just console.log. Would be better for the end developer.

Reply:
Hi Matt, I was trying to keep the scope of this work purely to translate the hooks into functional components and extract other bits of logic into hooks, I didn't want to change the existing logic whilst doing a big refactor such as that.